core/src/main/java/hudson/tasks/BatchFile.java

```diff
@@ -101,24 +100,28 @@ public String getDisplayName() {
          * Performs on-the-fly validation of the errorlevel.
          */
         @Restricted(DoNotUse.class)
-        public FormValidation doCheckUnstableReturn(@QueryParameter String value) {
-            value = Util.fixEmptyAndTrim(value);
-            if (value == null) {
-                return FormValidation.ok();
-            }
-            long unstableReturn;
-            try {
-                unstableReturn = Long.parseLong(value);
-            } catch (NumberFormatException e) {
-                return FormValidation.error(hudson.model.Messages.Hudson_NotANumber());
-            }
-            if (unstableReturn == 0) {
-                return FormValidation.warning(hudson.tasks.Messages.BatchFile_invalid_exit_code_zero());
-            }
-            if (unstableReturn < Integer.MIN_VALUE || unstableReturn > Integer.MAX_VALUE) {
-                return FormValidation.error(hudson.tasks.Messages.BatchFile_invalid_exit_code_range(unstableReturn));
-            }
-            return FormValidation.ok();
+        static public FormValidation doCheckUnstableReturn(@QueryParameter String value) {
```

Comment:
Why are we investing time in improving a method that shouldn't be used?:

`@Restricted(DoNotUse.class)`
_This type, field, or method shall never be referenced from anywhere at all._

Reply:
FTR the `@Restricted(DoNotUse.class)` is meant for Jelly calls. It's a method that is called by Stapler, detected using reflection during the form build process.

Reply:
Are you sure? I think it's used to warn programmers

Reply:
https://github.com/kohsuke/access-modifier/blob/master/access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/DoNotUse.java#L32